hub/core/chunk/chunk_compressed_chunk.py

```diff
@@ -1,123 +1,154 @@
+                self._changed = False
 
-            buffer_list.append(incoming_sample)
-            compressed_bytes = compress_multiple(buffer_list, self.compression)
-            if self.is_empty or len(compressed_bytes) < self.min_chunk_size:
-                self.data_bytes = compressed_bytes
-                # Byte positions aren't relevant for chunk wise img compressions
-                self.register_in_meta_and_headers(None, shape)
-                num_samples += 1
-                self._decompressed_samples = buffer_list
-            else:
-                buffer_list.pop()
-                break
+            shape = incoming_sample.shape
+            shape = self.normalize_shape(shape)
 
+            self.num_dims = self.num_dims or len(shape)
+            check_sample_shape(shape, self.num_dims)
+            self.decompressed_samples.append(incoming_sample)  # type: ignore
+            self._changed = True
+            # Byte positions are not relevant for chunk wise image compressions, so incoming_num_bytes=None.
+            self.register_in_meta_and_headers(None, shape)
+            num_samples += 1
         return num_samples
 
-    @property
-    def decompressed_samples(self) -> List[np.ndarray]:
-        """Applicable only for compressed chunks. Returns samples contained in this chunk as a list of numpy arrays."""
-        if self._decompressed_samples is None:
-            num_samples = self.shapes_encoder.num_samples
-            shapes = [self.shapes_encoder[i] for i in range(num_samples)]
-            self._decompressed_samples = decompress_multiple(self.data_bytes, shapes)
-        return self._decompressed_samples
-
-    @property
-    def decompressed_bytes(self) -> bytes:
-        """Applicable only for chunks compressed using a byte compression. Returns the contents of the chunk as a decompressed buffer."""
-        if self._decompressed_bytes is None:
-            try:
-                data = decompress_bytes(self.data_bytes, self.compression)
-                self._decompressed_bytes = data
-            except SampleDecompressionError:
-                raise ValueError(
-                    "Chunk.decompressed_bytes can not be called on chunks compressed with image compressions. Use Chunk.get_samples() instead."
-                )
-        return self._decompressed_bytes
-
     def read_sample(self, local_index: int, cast: bool = True, copy: bool = False):
         if self.is_image_compression:
-            return self.decompressed_samples[local_index]
+            return self.decompressed_samples[local_index]  # type: ignore
 
-        sb, eb = self.byte_positions_encoder[local_index]
         shape = self.shapes_encoder[local_index]
-        decompressed = memoryview(self.decompressed_bytes)
-        buffer = decompressed[sb:eb]
+        decompressed = memoryview(self.decompressed_bytes)  # type: ignore
+        if (
+            not self.byte_positions_encoder.is_empty()
```

Comment:
can we add to flag for is_tile to chunk classes instead? I feel like checking the current way could potentially lead to an edge case that we miss